File: app/src/main/java/com/github/overpass/gather/map/LocationPermissionUseCase.java
package com.github.overpass.gather.map;

import com.github.overpass.gather.SingleLiveEvent;

public class LocationPermissionUseCase {

    private final SingleLiveEvent<PermissionRequestResult> permissionRequestResultData;

    public LocationPermissionUseCase() {
        this.permissionRequestResultData = new SingleLiveEvent<>();
    }

    public SingleLiveEvent<PermissionRequestResult> getPermissionRequestResultData() {
        return permissionRequestResultData;
    }

    public void onRequestPermissionsResult(int requestCode,
                                    String[] permissions,
                                    int[] grantResults) {
        permissionRequestResultData.setValue(new PermissionRequestResult(requestCode,
                permissions, grantResults));
    }
}

File: app/src/main/java/by/overpass/gather/model/usecase/permission/LocationPermissionUseCase.kt
package by.overpass.gather.model.usecase.permission

import by.overpass.gather.ui.map.PermissionRequestResult
import com.hadilq.liveevent.LiveEvent

class LocationPermissionUseCase(
        val permissionRequestResultData: LiveEvent<PermissionRequestResult>
) {

    fun onRequestPermissionsResult(requestCode: Int,
                                   permissions: Array<String>,
                                   grantResults: IntArray) {
        permissionRequestResultData.value = PermissionRequestResult(requestCode,
                permissions, grantResults)
    }
}
